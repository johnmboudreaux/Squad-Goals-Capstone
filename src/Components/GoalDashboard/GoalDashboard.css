h1.goal-body-title {
  color: #d2d3d3;
  font-size: 2em;
  font-weight: lighter;
  margin: 0;
  height: 75px; }

.dashboard-body-info p {
  color: #d2d3d3;
  text-align: center;
  font-size: 2em;
  font-weight: lighter;
  width: 100%; }

.goal-dash-content {
  height: auto; }

.creator-info {
  height: 40px; }
