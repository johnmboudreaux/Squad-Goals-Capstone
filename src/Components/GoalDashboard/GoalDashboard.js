import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Conversation from '../Conversation/Conversation';
import SubmitComment from '../SubmitComment/SubmitComment';

class GoalDashboard extends Component {
  constructor() {
    super();
    this.state = {
      goal: {
        users: [],
      },
      activeUser: false,
    };
  }

  componentDidMount() {
    this.getGoalData();
  }

  getGoalData() {
    const url = `/api/v1/goals/${this.props.match.params.id}`;
    return this.contentFetch(url)
      .then((parsedData) => {
        this.setState({ goal: parsedData });
        this.checkForEnrolledUser();
      })
      .catch(error => console.error(error));
  }

  checkForEnrolledUser() {
    const { userId } = this.props.location.state;
    const goalId = parseInt(this.props.match.params.id);
    const url = `/api/v1/users/${userId}/goals/${goalId}`;
    return fetch(url)
      .then(result => result.json())
      .then((response) => {
        if (response.length === 1) {
          this.setState({ activeUser: true });
        }
      })
      .catch(error => console.error(error));
  }


  joinGoal() {
    const { userId } = this.props.location.state;
    const goalId = parseInt(this.props.match.params.id);
    const postBody = {
      user_id: userId,
      goal_id: goalId,
    };
    fetch(`/api/v1/users/${userId}/goals/${goalId}`, {
      method: 'POST',
      body: JSON.stringify(postBody),
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then(response => response)
      .catch(error => console.error(error));
  }

  leaveGoal(path1, id1) {
    const { userId } = this.props.location.state;
    const goalId = parseInt(this.props.match.params.id);
    fetch(`/api/v1/users/${userId}/goals/${goalId}`, {
      method: 'DELETE',
    })
      .then(response => response)
      .catch(error => console.error(error));
  }

  contentFetch(url) {
    return fetch(url)
      .then(result => result.json())
      .catch(error => console.error(error));
  }

  render() {
    return (
      <div className="dashboard-container">
        <div className="dashboard-body">
          <h1 className="dashboard-body-title goal-body-title">{this.state.goal.title}</h1>
          <div className="dashboard-body-info">
            <div className="squad-dashboard-body-titles">
              <p>{this.state.goal.description}</p>
            </div>
            <div className="goal-dash-content">
              <div>
                <h2>Starts at {this.state.goal.goal_time}</h2>
              </div>
              <div>
                <h2>{this.state.goal.goal_points}</h2>
              </div>
            </div>
            <div className="squad-dashboard-button">
              {
              this.state.activeUser ?
                <div className="single-button-container">
                  <input
                    className="nav-button"
                    onClick={() => this.leaveGoal()}
                    type="submit"
                    value="Leave"
                  />
                </div>
                :
                <div className="single-button-container">
                  <input
                    className="nav-button"
                    onClick={() => this.joinGoal()}
                    type="submit"
                    value="Join"
                  />
                </div>
              }
            </div>
          </div>
          { this.state.goal.conversation !== undefined &&
            <div className="conversation-container">
              <Conversation comments={this.state.goal.conversation} />
              <SubmitComment conversationId={this.state.goal.conversation_id} />
            </div>
          }
        </div>
      </div>
    );
  }
}

GoalDashboard.propTypes = {
  match: PropTypes.object,
  params: PropTypes.object,
  location: PropTypes.object,
  state: PropTypes.object,
  id: PropTypes.string,
};

export default GoalDashboard;
