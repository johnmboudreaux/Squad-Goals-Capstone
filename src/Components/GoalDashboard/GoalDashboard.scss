@import '../../variables.scss';


h1.goal-body-title {
  color: $white;
  font-size: 2em;
  font-weight: lighter;
  margin: 0;
  height: 75px;
}

.dashboard-body-info p {
  color: $white;
  text-align: center;
  font-size: 2em;
  font-weight: lighter;
  width: 100%
}

.goal-dash-content {
  height: auto;
}

.creator-info {
  height: 40px;
}
