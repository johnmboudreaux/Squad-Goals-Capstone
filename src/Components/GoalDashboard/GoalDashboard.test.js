import { shallow } from 'enzyme';
import React from 'react';
import GoalDashboard from './GoalDashboard';

describe('GoalDashboard snapshot', () => {
  const goal = {
    conversation: [],
    conversation_id: 24,
    id: 18,
    squad_name: 'Jm Squad Squad',
    users: [{}],
  };

  it('should always match the snapshot', () => {
    const wrapper = shallow(<GoalDashboard
      squad={goal}
      location={{ state: { userId: 1 } }}
      match={{ params: { id: 1 } }}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
