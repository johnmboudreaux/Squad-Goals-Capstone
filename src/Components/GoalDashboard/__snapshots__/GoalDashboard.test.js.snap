// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalDashboard snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoalDashboard
    location={
      Object {
        "state": Object {
          "userId": 1,
        },
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    squad={
      Object {
        "conversation": Array [],
        "conversation_id": 24,
        "id": 18,
        "squad_name": "Jm Squad Squad",
        "users": Array [
          Object {},
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="dashboard-body"
      >
        <h1
          className="dashboard-body-title goal-body-title"
        />
        <div
          className="dashboard-body-info"
        >
          <div
            className="squad-dashboard-body-titles"
          >
            <p />
          </div>
          <div
            className="goal-dash-content"
          >
            <div>
              <h2>
                Starts at 
              </h2>
            </div>
            <div>
              <h2 />
            </div>
          </div>
          <div
            className="squad-dashboard-button"
          >
            <div
              className="single-button-container"
            >
              <input
                className="nav-button"
                onClick={[Function]}
                type="submit"
                value="Join"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "dashboard-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="dashboard-body-title goal-body-title"
          />,
          <div
            className="dashboard-body-info"
          >
            <div
              className="squad-dashboard-body-titles"
            >
              <p />
            </div>
            <div
              className="goal-dash-content"
            >
              <div>
                <h2>
                  Starts at 
                </h2>
              </div>
              <div>
                <h2 />
              </div>
            </div>
            <div
              className="squad-dashboard-button"
            >
              <div
                className="single-button-container"
              >
                <input
                  className="nav-button"
                  onClick={[Function]}
                  type="submit"
                  value="Join"
                />
              </div>
            </div>
          </div>,
          false,
        ],
        "className": "dashboard-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "dashboard-body-title goal-body-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="squad-dashboard-body-titles"
              >
                <p />
              </div>,
              <div
                className="goal-dash-content"
              >
                <div>
                  <h2>
                    Starts at 
                  </h2>
                </div>
                <div>
                  <h2 />
                </div>
              </div>,
              <div
                className="squad-dashboard-button"
              >
                <div
                  className="single-button-container"
                >
                  <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join"
                  />
                </div>
              </div>,
            ],
            "className": "dashboard-body-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p />,
                "className": "squad-dashboard-body-titles",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h2>
                      Starts at 
                    </h2>
                  </div>,
                  <div>
                    <h2 />
                  </div>,
                ],
                "className": "goal-dash-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2>
                      Starts at 
                    </h2>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Starts at ",
                        undefined,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Starts at ",
                      undefined,
                    ],
                    "type": "h2",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2 />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h2",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="single-button-container"
                >
                  <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join"
                  />
                </div>,
                "className": "squad-dashboard-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join"
                  />,
                  "className": "single-button-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "nav-button",
                    "onClick": [Function],
                    "type": "submit",
                    "value": "Join",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="dashboard-body"
        >
          <h1
            className="dashboard-body-title goal-body-title"
          />
          <div
            className="dashboard-body-info"
          >
            <div
              className="squad-dashboard-body-titles"
            >
              <p />
            </div>
            <div
              className="goal-dash-content"
            >
              <div>
                <h2>
                  Starts at 
                </h2>
              </div>
              <div>
                <h2 />
              </div>
            </div>
            <div
              className="squad-dashboard-button"
            >
              <div
                className="single-button-container"
              >
                <input
                  className="nav-button"
                  onClick={[Function]}
                  type="submit"
                  value="Join"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "dashboard-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="dashboard-body-title goal-body-title"
            />,
            <div
              className="dashboard-body-info"
            >
              <div
                className="squad-dashboard-body-titles"
              >
                <p />
              </div>
              <div
                className="goal-dash-content"
              >
                <div>
                  <h2>
                    Starts at 
                  </h2>
                </div>
                <div>
                  <h2 />
                </div>
              </div>
              <div
                className="squad-dashboard-button"
              >
                <div
                  className="single-button-container"
                >
                  <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join"
                  />
                </div>
              </div>
            </div>,
            false,
          ],
          "className": "dashboard-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "dashboard-body-title goal-body-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="squad-dashboard-body-titles"
                >
                  <p />
                </div>,
                <div
                  className="goal-dash-content"
                >
                  <div>
                    <h2>
                      Starts at 
                    </h2>
                  </div>
                  <div>
                    <h2 />
                  </div>
                </div>,
                <div
                  className="squad-dashboard-button"
                >
                  <div
                    className="single-button-container"
                  >
                    <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join"
                    />
                  </div>
                </div>,
              ],
              "className": "dashboard-body-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p />,
                  "className": "squad-dashboard-body-titles",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h2>
                        Starts at 
                      </h2>
                    </div>,
                    <div>
                      <h2 />
                    </div>,
                  ],
                  "className": "goal-dash-content",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2>
                        Starts at 
                      </h2>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Starts at ",
                          undefined,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Starts at ",
                        undefined,
                      ],
                      "type": "h2",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2 />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h2",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="single-button-container"
                  >
                    <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join"
                    />
                  </div>,
                  "className": "squad-dashboard-button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join"
                    />,
                    "className": "single-button-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "nav-button",
                      "onClick": [Function],
                      "type": "submit",
                      "value": "Join",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
